packages/next-intl/src/navigation/shared/BaseLink.tsx

```diff
@@ -52,12 +52,12 @@ function BaseLink(
   }
 
   useEffect(() => {
-    if (!pathname || localePrefix === 'never') return;
+    if (!pathname) return;
 
     setLocalizedHref(
       localizeHref(href, locale, defaultLocale, pathname ?? undefined)
     );
-  }, [defaultLocale, href, locale, localePrefix, pathname]);
+  }, [defaultLocale, href, locale, pathname]);
```

Comment:
Why do we need this effect in the first place? Is this outdated and can we remove it?

Reply:
Added some notes to https://github.com/amannn/next-intl/issues/444#issue-1839328048, let's continue there …